frontend/src/lib/MlmdUtils.ts

```diff
@@ -211,7 +215,40 @@ export async function getOutputArtifactsInExecution(execution: Execution): Promi
     throw artifactsErr;
   }
 
-  return artifactsRes.getArtifactsList();
+  const artifactMap = artifactsRes.getArtifactsList().reduce((map, artifact) => {
+    map[artifact.getId()] = artifact;
+    return map;
+  }, {});
```

Comment:
nit for discussion: I personally prefer avoiding reduce, because it's not very intuitive, also it doesn't really improve readability much (compared to loops).

Other styles like using `Object.entries()` and `Object.fromEntries()` might make the code easier to read too.

But anyway, you can decide on this. Just saying some opinions.

Reply:
I think both approaches have similar effect/scalability, and I will gladly change to Object.entries() approach for readability.